src/main/java/formflow/library/config/SubflowManager.java

```diff
@@ -0,0 +1,201 @@
+package formflow.library.config;
+
+import formflow.library.data.Submission;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+@Component
+public class SubflowManager {
+
+    List<FlowConfiguration> flowConfigurations;
+    
+    public SubflowManager(List<FlowConfiguration> flowConfigurations) {
+        this.flowConfigurations = flowConfigurations;
+    }
+
+    public boolean subflowHasRelationship(String flow, String subflow) {
+        FlowConfiguration flowConfiguration = getFlowConfiguration(flow);
+        if (flowConfiguration.getSubflows().containsKey(subflow)) {
+            return flowConfiguration.getSubflows().get(subflow).getRelationship() != null;
+        } else {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+                    String.format("Subflow %s not found in flow %s. Check that your flows-config.yaml is configured correctly.", subflow, flow));
+        }
+    }
+    
+    public void addSubflowRelationshipData(ScreenNavigationConfiguration currentScreen, String flow, Submission submission) {
+        String subflowName = currentScreen.getSubflow();
+        if (!subflowHasRelationship(flow, subflowName)) {
+            // No relationship so we don't need to do anything
+            return;
+        }
+
+        SubflowConfiguration currentSubflow = getSubflowConfiguration(flow, subflowName);
+        String relatedSubflowName = currentSubflow.getRelationship().getRelatesTo();
+        String relatedIdKey = currentSubflow.getRelationship().getRelationAlias();
+
+        List<Map<String, Object>> relatedSubflowData = getSubflowData(submission, relatedSubflowName);
+        List<Map<String, Object>> currentSubflowData = getOrCreateSubflowData(submission, subflowName);
+
+        if (!submission.getInputData().containsKey(subflowName)) {
+            // Initial setup: add all related items as incomplete iterations
+            relatedSubflowData.forEach(relatedItem ->
+                    currentSubflowData.add(createSubflowIterationWithRelationship(relatedIdKey, relatedItem.get("uuid")))
+            );
+        } else if (currentSubflowData.size() != relatedSubflowData.size()) {
```

Comment:
I am a bit stuck on this one:

relatedSubflowsData size will not change
but currentSubflowsData size can change.

This method is called when the model is created. So the first time if is called, the data would not be set.

It would be in subsequent times that this statement could be true.

Why do we need to keep calling the addSubFlowRelationship data flow instead of adding this check in the model method?

if (!submission.getInputData().containsKey(subflowName)) {